src/arbitrum/epoch-based-managed-vaults/stratagies/Y2KFinanceStrategy.sol

```diff
@@ -1,7 +1,126 @@
+
+    function isActive() public view override returns (bool) {
+        if (isStrategiesActiveOverride) return false;
+        if (IERC20(assetVaultPrimaryAsset).balanceOf(address(this)) > 0) return true;
+
+        address[] memory _vaults = vaults;
+        uint256[] memory _vaultIDs = vaultIDs;
+        for (uint256 i = 0; i < vaults.length; i++) {
+            if (IY2KVault(_vaults[i]).balanceOf(address(this), _vaultIDs[i]) > 0) return true;
+        }
+
+        return false;
+    }
+
+    /********************************** Manager Functions **********************************/
+
+    /// @dev Executes the strategy - deposit into a Y2K Risk/Hedge Vault
+    /// @dev _configData:
+    /// @dev _index - index of the vault in the vaultFactory, determines the asset and the strike vault
+    /// @dev _amount - amount of the assetVaultPrimaryAsset to deposit, will be set to the balance if set to type(uint256).max
+    /// @dev _id - id of the vault in the vaultFactory, determines end date of the Epoch
+    /// @dev _type - true for Risk Vault, false for Hedge Vault
+    function execute(bytes memory _configData) external override onlyManager returns (uint256) {
+        
+        (uint256 _id, uint256 _amount, uint256 _before, address _vault) = _getConfig(_configData);
+
+        _approve(assetVaultPrimaryAsset, _vault, _amount);
+        IY2KVault(_vault).deposit(_id, _amount, address(this));
+
+        uint256 _shares = IY2KVault(_vault).balanceOf(address(this), _id) - _before;
+
+        // TODO
+        // if (_getStakingContract(_vault, _id) != address(0)) _stake(_shares);
+
+        return _shares;
+    }
+
+    /// @dev Terminates the strategy - withdraw from fortGLP
+    /// @dev _configData:
+    /// @dev _index - index of the vault in the vaultFactory, determines the asset and the strike vault
+    /// @dev _amount - amount of the assetVaultPrimaryAsset to deposit, will be set to the balance if set to type(uint256).max
+    /// @dev _id - id of the vault in the vaultFactory, determines end date of the Epoch
+    /// @dev _type - true for Risk Vault, false for Hedge Vault
+    function terminate(bytes memory _configData) external override onlyManager returns (uint256) {
+        
+        (uint256 _id, uint256 _amount, uint256 _before, address _vault) = _getConfig(_configData);
+
+        // TODO
+        // if (_getStakingContract(_vault, _id) != address(0)) _unstake(_shares);
+
+        IY2KVault(_vault).withdraw(_id, _amount, address(this), address(this));
+
+        return IY2KVault(_vault).balanceOf(address(this), _id) - _before;
+    }
+
+    /********************************** Internal Functions **********************************/
+
+    function _getConfig(bytes memory _configData) internal returns (uint256, uint256, uint256, address) {
+        (uint256 _index, uint256 _amount, uint256 _id, bool _type) = abi.decode(_configData, (uint256, uint256, uint256, bool));
```

Comment:
Can we document how the RPC call expects the config data bytes? Is it basically just some JSON object with the respective field names and values?

Reply:
Oh yeah. We'll need good docs for this one. It's basically a bunch of data that you encode and pass to the contract, no biggie.